### PR TITLE
Gloo Workers v2

### PR DESCRIPTION
**Pull request by droogmic**
This currently fails with:

```
SyntaxError: WorkerGlobalScope.importScripts: Failed to load worker script at "worker.js"
```

### CODE REVIEW

**Comment by droogmic** [comment]
This now works, but it depends on an upstream branch I don't control.

File: osm2lanes/src/test.rs [hunk 1]
```diff
@@ -478,6 +478,31 @@ mod tests {
         });
     }
 
+    #[test]
+    fn test_json() {
```
Comment (dabreegster) [inline osm2lanes/src/test.rs:482]: This was motivated by tracking down where the serde flattening bug was happening, right?

**Review by dabreegster — APPROVED** [review]
This appears to work nice and quickly locally, and the code isn't hard to follow! I guess the only concern was `gloo-worker = { git = "https://github.com/futursolo/gloo", rev = "a2101e166260294292c8121fdb8ed883dae62ed8" }`, right? Unless the branch is deleted, there shouldn't be any problems with reproducible builds.

Awesome work!

File: osm2lanes/src/test.rs (hunk 1 above)
Comment (droogmic) [reply osm2lanes/src/test.rs:482]: yeah, if you replace this with bincode, the test fails :)
